server/logger.js

```diff
@@ -0,0 +1,100 @@
+import winston from 'winston';
+import fs from 'fs';
+import { fileURLToPath } from 'url';
+import path from 'path';
+
+const { combine, timestamp, printf, colorize } = winston.format;
```

Comment:
The `updateSocket` function should validate the `user_id` parameter to ensure it is not null or undefined before assigning it to `socket_user_map`. This will help prevent potential issues with invalid user IDs.
```diff
+  if (!user_id) {
+    throw new Error('Invalid user_id');
+  }
```


    *This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*